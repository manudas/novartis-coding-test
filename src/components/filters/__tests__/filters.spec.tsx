import React from 'react'

import {
    fireEvent,
    render,
    screen,
    within,
} from '@testing-library/react'

import { Filters } from '..'

describe('Filters test suite', () => {

    it('should render <Filters />', () => {
        const setGeneSymbol = jest.fn()
        const setLineage = jest.fn()

        const { container } = render(
            <Filters
                geneSymbol={'TP53'}
                lineage={'lung'}
                changeGeneSymbolHandler={({
                    target: {
                        value
                    }
                }) => {
                    setGeneSymbol(value)
                }}
                changeLineageHandler={({
                    target: {
                        value
                    }
                }) => {
                    setLineage(value)
                }}
            />
        )
        expect(container).toMatchSnapshot()
    })

    it('should call setGeneSymbol when dropdown value changes in first filter', () => {
        const setGeneSymbol = jest.fn()
        const setLineage = jest.fn()

        render(
            <Filters
                geneSymbol={'TP53'}
                lineage={'lung'}
                changeGeneSymbolHandler={({
                    target: {
                        value
                    }
                }) => {
                    setGeneSymbol(value)
                }}
                changeLineageHandler={({
                    target: {
                        value
                    }
                }) => {
                    setLineage(value)
                }}
            />
        )

        // First filter is Gene filter, second is Lineage
        const filters = screen.getAllByRole('button')
        fireEvent.mouseDown(filters[0]) // gene symbol

        const geneFilterOptions = within(screen.getByRole('listbox'))
        fireEvent.click(geneFilterOptions.getByText(/kras/i))
        expect(setGeneSymbol).toHaveBeenCalledWith('KRAS')
    })

    it('should call setLineage when dropdown value changes in second filter', () => {
        const setGeneSymbol = jest.fn()
        const setLineage = jest.fn()

        render(
            <Filters
                geneSymbol={'TP53'}
                lineage={'lung'}
                changeGeneSymbolHandler={({
                    target: {
                        value
                    }
                }) => {
                    setGeneSymbol(value)
                }}
                changeLineageHandler={({
                    target: {
                        value
                    }
                }) => {
                    setLineage(value)
                }}
            />
        )

        // First filter is Gene filter, second is Lineage
        const filters = screen.getAllByRole('button')
        fireEvent.mouseDown(filters[1]) // lineage

        const lineageFilterOptions = within(screen.getByRole('listbox'))
        fireEvent.click(lineageFilterOptions.getByText(/blood/i))
        expect(setLineage).toHaveBeenCalledWith('blood')
    })
})