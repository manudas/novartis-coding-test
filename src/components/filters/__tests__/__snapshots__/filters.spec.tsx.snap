// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters test suite should render <Filters /> 1`] = `
<div>
  <header
    class="App-filters"
  >
    <div
      class="MuiFormControl-root App-filter css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        id="gene-symbol-label"
      >
        Gene Symbol
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="gene-symbol-label gene-symbol"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="gene-symbol"
          role="button"
          tabindex="0"
        >
          TP53
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="TP53"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Gene Symbol
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root App-filter css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        id="lineage-label"
      >
        Lineage
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="lineage-label lineage"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="lineage"
          role="button"
          tabindex="0"
        >
          lung
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="lung"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Lineage
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </header>
</div>
`;
